src/Microsoft.DotNet.Wpf/src/ReachFramework/Packaging/XpsFixedDocumentReaderWriter.cs

```diff
@@ -1,1278 +1,1282 @@
+        /// package using the given URI.
+        /// </summary>
+        /// <param name="pageUri">
+        /// The URI of the fixed page to retrieve.
+        /// </param>
+        /// <returns>
+        /// Returns an interface to an IXpsFixedPageReader to read
+        /// the requested fixed page.
+        /// </returns>
+        public
+        IXpsFixedPageReader
+        GetFixedPage(
+            Uri         pageUri
+            )
+        {
+            UpdatePageCache();
+            IXpsFixedPageReader pageReader = null;
+
+            foreach (IXpsFixedPageReader reader in _pageCache )
+            {
+                if( reader.Uri == pageUri )
+                {
+                    pageReader =  reader;
+                }
+            }
+
+            return pageReader;
+        }
+
+        public
+        void
+        AddSignatureDefinition(
+            XpsSignatureDefinition signatureDefinition
+            )
+        {
+            EnsureSignatureDefinitions();
+            _signatureDefinitions.Add( signatureDefinition );
+            _sigCollectionDirty = true;
+}
+
+        public
+        void
+        CommitSignatureDefinition()
+        {
+            bool isDirty = false;
+
+            //
+            // if the collection is dirty not point in testing
+            // each signature.
+            //
+            if( !_sigCollectionDirty )
+            {
+                foreach( XpsSignatureDefinition sigDef in _signatureDefinitions )
+                {
+                    if( sigDef.HasBeenModified )
+                    {
+                        isDirty = true;
+                        break;
+                    }
+                }
+            }
+            if( isDirty || _sigCollectionDirty )
+            {
+                WriteSignatureDefinitions();
+            }
+        }
+
+        /// <summary>
+        /// This method removes a signature definitions associated with
+        /// the FixedDocument
+        /// </summary>
+        /// <param name="signatureDefinition">
+        /// Signature Definition to remove
+        /// </param>
+        public
+        void
+        RemoveSignatureDefinition(XpsSignatureDefinition signatureDefinition)
+        {
+           EnsureSignatureDefinitions();
+           _signatureDefinitions.Remove( signatureDefinition );
+           _sigCollectionDirty = true;
+}
+
+        /// <summary>
+        /// This method commits any changes not already committed for this
+        /// fixed document.
+        ///
+        /// NOTE:  This commits changes to all child object under this
+        /// branch of the tree.  No further changes will be allowed.
+        /// </summary>
+        public
+        void
+        Commit(
+            )
+        {
+            CommitInternal();
+        }
+
+        /// <summary>
+        /// This method closes streams and frees memory for this
+        /// fixed document.
+        ///
```

Comment:
While we're swapping to usings, maybe we should add one for the stream and writer here?

Reply:
good idea.